File: Is it Fibonacci.cpp
class Solution {
  public:
    long long solve(int N, int K, vector<long long> GeekNum) {
        // code here
        long long sum = 0;
        vector<long long> vec;
        

        for (auto it : GeekNum) sum += it, vec.push_back(it);

        int beg = 0;
        while (vec.size() < N) {
            vec.push_back(sum);
            sum -= vec[beg];
            sum += vec.back();
            beg++;
        }

        return vec[N - 1];
    }
};

File: Is it Fibonacci.py
class Solution():
    def solve(self, N, K, GeekNum):
        sumo = 0
        for i in GeekNum:
            sumo += i
        beg = 0
        while len(GeekNum) < N:
            GeekNum.append(sumo)
            sumo -= GeekNum[beg]
            sumo += GeekNum[-1]
            beg += 1
        return GeekNum[N-1]
